### ISSUE TITLE
Document.populate & Document.execPopulate in a transaction

### ISSUE THREAD

**Issue by sebastian-nowak** [Automattic/mongoose#6754]
Docs for `doc.$session()` only mention that it applies `save()`:

<img width="871" alt="screen shot 2018-07-22 at 16 35 52" src="https://user-images.githubusercontent.com/5895458/43046724-612fd982-8dcd-11e8-98e8-114e3fb2c3bd.png">

Does it mean that calling `doc.populate()` or `doc.execPopulate` will ignore the associated mongo session?

**Comment by vkarpov15** [comment]
Yes it will, we missed that. Thanks for pointing out this issue, we will fix ASAP :+1:

**Comment by vkarpov15** [comment]
Fixed in master, will be released in 5.2.6 next Monday :+1:

**Comment by sebastian-nowak** [comment]
That's great, thank you @vkarpov15 🙂

Just noticed one thing:

```js
     // Use `$session()` by default if the document has an associated session
     // See gh-6754
     if (this.$session() != null) {
       const session = this.$session();
       paths.forEach(path => {
         path.options = path.options || {};
         path.options.session = session;
       });
     }
```

It means that `$session` on a document takes precedence over session passed explicitly via options. Shouldn't it be the other way around?
